### PR TITLE
Add possibility to upload to Harbor

### PR DESCRIPTION
**Pull request by torrespro**
doc: update Readme with the new upload to Harbor (via Harbor API)
feature: add Harbor to he UploadMojo (https://github.com/kiwigrid/helm-maven-plugin/issues/83)
feature: upload to Nexus, Harbor and Chartmuseum now support username/pwd from settings.xml file https://github.com/kiwigrid/helm-maven-plugin/issues/105

### CODE REVIEW

File: src/main/java/com/kiwigrid/helm/maven/plugin/UploadMojo.java [hunk 1]
```diff
@@ -73,34 +86,40 @@ protected void uploadSingle(String file) throws IOException, BadUploadException,
 		case NEXUS:
 			connection = getConnectionForUploadToNexus(fileToUpload);
 			break;
+        case HARBOR:
+			uploadToHarbor(fileToUpload);
+			break;
```
Comment (jjjasper) [inline src/main/java/com/kiwigrid/helm/maven/plugin/UploadMojo.java:91]: Alternatively, `return` and have no need for the very big conditional block `if (uploadRepo.getType() != RepoType.HARBOR)`

File: src/main/java/com/kiwigrid/helm/maven/plugin/UploadMojo.java [hunk 2]
```diff
@@ -73,34 +86,40 @@ protected void uploadSingle(String file) throws IOException, BadUploadException,
...
+				String response;
+				if (connection.getErrorStream() != null) {
+					response = IOUtils.toString(connection.getErrorStream(), Charset.defaultCharset());
+				} else if (connection.getInputStream() != null) {
+					response = IOUtils.toString(connection.getInputStream(), Charset.defaultCharset());
+				} else {
+					response = "No details provided";
+				}
+				throw new BadUploadException(response);
+			} else {
```
Comment (jjjasper) [inline src/main/java/com/kiwigrid/helm/maven/plugin/UploadMojo.java:111]: `if` block ends with exception, no need for `else`.

**Comment by grdaneault** [comment]
Hi! Any chance of an update here?

File: src/main/java/com/kiwigrid/helm/maven/plugin/UploadMojo.java [hunk 3]
```diff
@@ -154,6 +195,33 @@ protected HttpURLConnection getConnectionForUploadToNexus(File file) throws IOEx
 		return connection;
 	}
 
+	protected void uploadToHarbor(File file) throws IOException, BadUploadException, MojoExecutionException {
```
Comment (axdotl) [inline src/main/java/com/kiwigrid/helm/maven/plugin/UploadMojo.java:198]: Upload to Harbor is implemented different from other uploads.

I would prefer to get a method like
```
protected HttpURLConnection getConnectionForUploadToHarbor() throws IOException {
}
```

What do you think?

**Review by axdotl — CHANGES_REQUESTED** [review]

**Comment by sschnabe** [comment]
Closing due of age.

**Pull request closed without merging**